uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/carddetailscheckout/mapper/CardCheckOutFullCardPaymentPayloadMapper.kt

```diff
@@ -0,0 +1,44 @@
+package tech.dojo.pay.uisdk.presentation.ui.carddetailscheckout.mapper
+
+import tech.dojo.pay.sdk.card.entities.DojoAddressDetails
+import tech.dojo.pay.sdk.card.entities.DojoCardDetails
+import tech.dojo.pay.sdk.card.entities.DojoCardPaymentPayLoad
+import tech.dojo.pay.uisdk.presentation.ui.carddetailscheckout.state.CardDetailsCheckoutState
+
+internal class CardCheckOutFullCardPaymentPayloadMapper {
+
+    fun getPaymentPayLoad(
+        currentState: CardDetailsCheckoutState,
+        isStartDestination: Boolean,
+    ): DojoCardPaymentPayLoad.FullCardPaymentPayload =
```

Comment:
I'm not sure about having the state to make the mapper here 🤔 
but how it was structured I don't see much options. I'm adding this comment in case any other person see how to improve it